crates/miden-lib/src/account/components/mod.rs

```diff
@@ -191,20 +191,17 @@ impl WellKnownComponent {
     /// interface to the component interface vector.
     fn extract_component(
         &self,
-        procedures_map: &mut BTreeMap<Word, &AccountProcedureInfo>,
+        procedures_map: &mut BTreeMap<Word, &AccountProcedureRoot>,
```

Comment:
The key and value in this map are basically the same, right? If so, shouldn't this be just `BTreeSet<AccountProcedureRoot>`?